File: JavaKotlinDers16WorkManager/app/src/main/java/com/example/javakotlinders16workmanager/RefreshDatabase.java
package com.example.javakotlinders16workmanager;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.annotation.NonNull;
import androidx.work.Data;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class RefreshDatabase extends Worker {

    Context myContext;

    public RefreshDatabase(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
        this.myContext=context;
    }

    @NonNull
    @Override
    public Result doWork() {//workmanager'ın ne yapacağını bu metot altına yapıyoruz.
        Data data=getInputData();//bize bir girdi verilecek onu al demek
        int myNumber=data.getInt("intKey",0);
        refreshDatabase(myNumber);

        return Result.success();
    }

    private void refreshDatabase(int myNumber){
        SharedPreferences sharedPreferences=myContext.getSharedPreferences("com.example.javakotlinders16workmanager",Context.MODE_PRIVATE);
        int mySavedNumber=sharedPreferences.getInt("mynumber",0);
        mySavedNumber+=myNumber;
        sharedPreferences.edit().putInt("mynumber",mySavedNumber).apply();
    }
}

File: KotlinDers15WorkManager/app/src/main/java/com/example/kotlinders15workmanager/RefreshDatabase.kt
package com.example.kotlinders15workmanager

import android.content.Context
import androidx.work.Worker
import androidx.work.WorkerParameters

/**
 * amacımız bunu uygulama kapalıyken de çalıştırmak
 * Mainden buraya bir veri göndermek istiyorsak: val getData=inputData kullanıyoruz
 *
 */
class RefreshDatabase(val context: Context, workerParams: WorkerParameters) : Worker(context,
    workerParams
) {

    override fun doWork(): Result {
        val getData=inputData
        val myNumber=getData.getInt("intKey",0)

        refreshDatabase(myNumber)
        return Result.success()
    }

    private fun refreshDatabase(myNumber: Int){
        val sharedPreferences=context.getSharedPreferences("com.example.kotlinders15workmanager",Context.MODE_PRIVATE)
        var mySavedNumber=sharedPreferences.getInt("myNumber",0)
        mySavedNumber+=myNumber
        println(mySavedNumber)
        sharedPreferences.edit().putInt("myNumber",mySavedNumber).apply()

    }
}